src/modules/land_detector/MulticopterLandDetector.cpp

```diff
@@ -154,40 +149,88 @@ bool MulticopterLandDetector::_get_ground_contact_state()
 		return true;
 	}
 
+	const bool lpos_available = (hrt_elapsed_time(&_vehicle_local_position.timestamp) < 1_s);
+
 	// land speed threshold
-	float land_speed_threshold = 0.9f * math::max(_params.landSpeed, 0.1f);
+	const float land_speed_threshold = 0.9f * math::max(_params.landSpeed, 0.1f);
+
 
-	// Check if we are moving vertically - this might see a spike after arming due to
-	// throttle-up vibration. If accelerating fast the throttle thresholds will still give
-	// an accurate in-air indication.
 	bool vertical_movement = false;
 
-	if (hrt_elapsed_time(&_landed_time) < LAND_DETECTOR_LAND_PHASE_TIME_US) {
+	if (lpos_available && _vehicle_local_position.v_z_valid) {
+		// Check if we are moving vertically - this might see a spike after arming due to
+		// throttle-up vibration. If accelerating fast the throttle thresholds will still give
+		// an accurate in-air indication.
+		float max_climb_rate = math::min(land_speed_threshold * 0.5f, _param_lndmc_z_vel_max.get());
```

Comment:
```suggestion
	float max_climb_rate;
	if (lpos_available && _vehicle_local_position.v_z_valid) {
		// Check if we are moving vertically - this might see a spike after arming due to
		// throttle-up vibration. If accelerating fast the throttle thresholds will still give
		// an accurate in-air indication.
		max_climb_rate = math::min(land_speed_threshold * 0.5f, _param_lndmc_z_vel_max.get());
```
with the next suggestion, I think it more clear. and reduce non-needed calculation